sql/core/src/main/scala/org/apache/spark/sql/execution/debug/package.scala

```diff
@@ -116,6 +175,30 @@ package object debug {
     }
   }
 
+  implicit class DebugStreamQuery(query: StreamingQuery) extends Logging {
+    def debug(): Unit = {
+      unwrapStreamingQueryWrapper(query) match {
+        case w: StreamExecution =>
+          if (w.lastExecution == null) {
+            debugPrint("No physical plan. Waiting for data.")
+          } else {
+            val executedPlan = w.lastExecution.executedPlan
+            if (executedPlan.find(_.isInstanceOf[WriteToContinuousDataSourceExec]).isDefined) {
+              debugPrint("Debug on continuous mode is not supported.")
```

Comment:
What's the issue you hit here?

Reply:
`debug()` will run query shortly to collect information: for micro-batch it will be 1 batch but in continuous mode it will never be finished.